### DISCUSSION TITLE
How to add regex expression for query params in source key in the rewrites function in next.config.js ?

### DISCUSSION THREAD

**Discussion by talktopreeti** [vercel/next.js#21770] [Help]
I wanted to check if next.js allows regex expression for query params present in source path in rewrites function in next.config.js.
I am facing issue in  following scenario as mentioned below - 

Requirement - Only the home page url ie "/" should reroute to the page present in Next.js server and home page url with query params should route/proxy to another server **along with the query params** present in home page url .

Example:
1. Incoming request is http://localhost:3000 , This should route to home page in next.config.js . I am able to achieve this routing by adding below rewrite in next.config.js
            {
            source: '/',
            destination: '/home'
            }
2. Incoming request is http://localhost:3000/?search=12345 , this should route to backend server along with query params eg http://<anotherserver>/?search=12345 . I am **not** able to achieve this routing with rewrite option in next.config.js. Have tried lot of regex options etc. In some of them I get an issue stating that the source pattern cannot begin with "?" .
            incoming url : 'http://localhost:3000/?search=123456',
            destination url :  http://<process.env.PROXY_URL>/?search=123456

Any help on this is highly appreciated.

**Comment by gaubansa3** [comment]
hey,,have you received any solution for 2nd part where matching query params in the source
